extensions/widgets/switchbutton/switchbutton.lcb

```diff
@@ -17,199 +17,709 @@ along with LiveCode.  If not see <http://www.gnu.org/licenses/>.  */
+	put "251,63,28" into mAndroidLightOnColors["Deep Orange"]
+	put "101,67,56" into mAndroidLightOnColors["Brown"]
+	put "140,140,140" into mAndroidLightOnColors["Grey"]
+	put "78,106,120" into mAndroidLightOnColors["Blue Grey"]
+
+	put the empty array into mAndroidDarkOnColors
+	put "233,133,136" into mAndroidDarkOnColors["Red"]
+	put "239,119,162" into mAndroidDarkOnColors["Pink"]
+	put "193,124,206" into mAndroidDarkOnColors["Purple"]
+	put "163,135,210" into mAndroidDarkOnColors["Deep Purple"]
+	put "141,149,209" into mAndroidDarkOnColors["Indigo"]
+	put "127,190,247" into mAndroidDarkOnColors["Blue"]
+	put "112,201,248" into mAndroidDarkOnColors["Light Blue"]
+	put "113,215,229" into mAndroidDarkOnColors["Cyan"]
+	put "112,194,184" into mAndroidDarkOnColors["Teal"]
+	put "150,207,151" into mAndroidDarkOnColors["Green"]
+	put "186,220,148" into mAndroidDarkOnColors["Light Green"]
+	put "224,237,138" into mAndroidDarkOnColors["Lime"]
+	put "255,245,140" into mAndroidDarkOnColors["Yellow"]
+	put "254,219,111" into mAndroidDarkOnColors["Amber"]
+	put "254,193,109" into mAndroidDarkOnColors["Orange"]
+	put "253,153,127" into mAndroidDarkOnColors["Deep Orange"]
+	put "174,154,147" into mAndroidDarkOnColors["Brown"]
+	put "234,234,234" into mAndroidDarkOnColors["Grey"]
+	put "160,176,185" into mAndroidDarkOnColors["Blue Grey"]
+
+	put color [249/255,249/255,249/255] into mAndroidLightThumbOff
+	put color [175/255,175/255,175/255] into mAndroidLightThumbDisabled
+	put color [0,0,0,0.26] into mAndroidLightTrackOff
+	put color [0,0,0,0.12] into mAndroidLightTrackDisabled
+
+	put color [175/255,175/255,175/255] into mAndroidDarkThumbOff
+	put color [51/255,51/255,51/255] into mAndroidDarkThumbDisabled
+	put color [1,1,1,0.3] into mAndroidDarkTrackOff
+	put color [1,1,1,0.1] into mAndroidDarkTrackDisabled
+
+end handler
+
+private handler updateVariables() returns nothing
+	put my width into mWidth
+	put my height into mHeight
+	put mWidth/9 into mRadius
+end handler
+
+private handler clickableRect() returns Rectangle
+	if mWidgetTheme is "iOS" then
+		return rectangle [mWidth/2 - mHeight/2, mHeight/4, mWidth/2 + mHeight/2, 3*mHeight/4]
+	else if mWidgetTheme contains "Android" then
+		return rectangle [1.5*mRadius, mHeight/2 - mRadius, 6*mRadius, mHeight/2 + mRadius]
 	end if
 end handler
 
-public handler OnMouseUp()
-	if the click position is within getRect("bounding box") then
-		setChecked(not(mIsChecked))
-		put "not animating" into mState
+public handler OnMouseMove() returns nothing
+	if mIsPressed then
+		variable tXMousePos as Real
+		variable tChange as Real
+		variable tMovedRight as Boolean
+
+		put true into mMouseHasMoved
+
+		put the x of the mouse position into tXMousePos
+		put tXMousePos - mXClick into tChange
+		if tChange > 0 then
+			put true into tMovedRight
+		else
+			put false into tMovedRight
+		end if
+
+		if tMovedRight then
+			put true into mSwitchIsOn
+		else
+			put false into mSwitchIsOn
+		end if
```

Comment:
I think you need to use a separate variable here rather than `mSwitchIsOn` which you then copy to `mSwitchIsOn` on mouseUp - otherwise when the mouse is down and the moved to the on position, if `put the switchIsOn of widget "Button"` is executed it will say true, even though the mouse hasn't been released